packages/@aws-cdk/core/lib/private/tree-metadata.ts

```diff
@@ -59,9 +66,16 @@ export class TreeMetadata extends Construct {
       version: 'tree-0.1',
       tree: visit(this.node.root),
     };
+    this._tree = lookup;
```

Comment:
Setting an instance field inside a method. It open the door for using stale data, in case that method is invoked again and changes the field (which cannot happen if done inside the constructor)